vm/policy.go

```diff
@@ -107,59 +110,71 @@ func PolicyHandling(t string, path string, o PolicyOptions, httpAddress string,
 		return err
 	}
 
-	js, err := yaml.YAMLToJSON(policyFile)
-	if err != nil {
-		return err
-	}
-
-	err = json.Unmarshal(js, &k)
-	if err != nil {
-		return err
-	}
+	policies := strings.Split(string(policyFile), "---")
 
-	var hostPolicy tp.K8sKubeArmorHostPolicy
-	var networkPolicy v2.CiliumNetworkPolicy
-	var policyEvent interface{}
+	for _, policy := range policies {
+		if policy == "" {
+			continue
+		}
 
-	if k.Kind == KubeArmorHostPolicy {
-		err = json.Unmarshal(js, &hostPolicy)
+		js, err := yaml.YAMLToJSON([]byte(policy))
 		if err != nil {
 			return err
 		}
 
-		policyEvent = tp.K8sKubeArmorHostPolicyEvent{
-			Type:   t,
-			Object: hostPolicy,
-		}
-
-	} else if k.Kind == CiliumNetworkPolicy {
-		err = json.Unmarshal(js, &networkPolicy)
+		err = json.Unmarshal(js, &k)
 		if err != nil {
 			return err
 		}
 
-		policyEvent = NetworkPolicyRequest{
-			Type:   t,
-			Object: networkPolicy,
-		}
+		var hostPolicy tp.K8sKubeArmorHostPolicy
+		var networkPolicy v2.CiliumNetworkPolicy
+		var policyEvent interface{}
 
-	}
+		if k.Kind == KubeArmorHostPolicy {
+			err = json.Unmarshal(js, &hostPolicy)
+			if err != nil {
+				return err
```

Comment:
error handling in this function is tricky!
What happens if one of the policy unmarshal fails and other were ok. All the policies before the failed policies were applied.

Reply:
Instead we can log the error and continue.

Reply:
> All the policies before the failed policies were applied.

 @nyrahul Yes. This is similar to how `kubectl` works. If we have 5 policies, out of which the first three policies are good and the fourth one is invalid, then `kubectl` will apply the first three policies successfully and throw error while processing the fourth one.

Reply:
@seswarrajan The error returned from this function will be printed in the user console.